src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/Utility/InstalledPackageCache.cs

```diff
@@ -0,0 +1,34 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using NuGet.Packaging.Core;
+using NuGet.Versioning;
+
+namespace NuGet.PackageManagement.VisualStudio.Utility
+{
+    internal class InstalledPackageCache
+    {
+        public readonly PackageIdentity _installedPackage;
```

Comment:
You can use: 

```suggestion
        public VersionRange AllowedVersions { get; }
```

It basically automatically creates a readonly prop.